tests/test_rtdc_feat_anc_plugin.py

```diff
@@ -903,7 +904,88 @@ def test_pf_wrong_length_2():
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "image_gauss_filter" in events.keys()
+        feat = h5["events"]['image_gauss_filter']
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_non_scalar_plugin_data_without_shape_issue_171():
+    """Test storage of 3D image plugin feature attributes without shape info
+     in recipe"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_non_scalar_feature()
+    PlugInFeature("image_gauss_filter", info)
+    with dclab.new_dataset(h5path) as ds:
+        image_gauss_filter = gaussian_filter(ds["image"], sigma=(0, 1, 1))
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("image_gauss_filter", image_gauss_filter)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "image_gauss_filter" in events.keys()
+        feat = h5["events"]['image_gauss_filter']
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_scalar_plugin_data_with_shape_issue_171():
+    """Test storage of scalar plugin feature attributes with shape info in
+    recipe"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    with dclab.new_dataset(h5path) as ds:
+        circ_per_area = ds["circ"] / ds["area_um"]
+    info = example_plugin_info_single_feature()
+    # Adding feature shape to the info
+    info["feature shapes"] = [circ_per_area.shape]
+    PlugInFeature("circ_per_area", info)
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("circ_per_area", circ_per_area)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "circ_per_area" in events.keys()
+        feat = h5["events"]['circ_per_area']
+        assert b'CLASS' not in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' not in feat.attrs.keys()
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_scalar_plugin_data_without_shape_issue_171():
+    """Test storage of scalar plugin feature attributes with shape info in
+    recipe"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    with dclab.new_dataset(h5path) as ds:
+        circ_per_area = ds["circ"] / ds["area_um"]
+    info = example_plugin_info_single_feature()
+    PlugInFeature("circ_per_area", info)
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("circ_per_area", circ_per_area)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "circ_per_area" in events.keys()
+        feat = h5["events"]['circ_per_area']
+        assert b'CLASS' not in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' not in feat.attrs.keys()
```

Comment:
You can definietly add more tests. Paul mentioned 4 possible tests in his https://github.com/ZELLMECHANIK-DRESDEN/dclab/pull/172#issuecomment-1080494432. You can find more edge cases I think!

What happens when you give 4D? are the correct errors raised? What happens if I save an image and define the shape of a contour. Is that possible? If it isn't, show me where :)

Reply:
Yes, you are right. I have to add more tests for every possibility. Right now, I am doing it locally. While doing so, I found that I had a problem with contour data because of its shape `[len, nan, 2]`. I guess we have to make conditions in the `RTDCWriter.store_features()` method to handle every situation (like contour, 2darray, image, 4darray, mask, scalar) of plugin feature. Right now, I just added conditions for image (3D) data only.